### PR TITLE
Frameworks and Targets update

### LINKED ISSUE #1694: Add more information to the Frameworks and Targets topic
Topic: https://docs.microsoft.com/en-us/dotnet/articles/standard/frameworks

Proposal:
Rename that topic to Target Frameworks
Add list of supported TFMs (similar to what we have in this topic over in the [NuGet docs](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks))
Link to this topic where we mention the target framework, TFMs, etc. For example, a bunch of commands like dotnet-build, dotnet-run have the framework option.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #1694 

* Topic renamed to *Target Frameworks*
* NuGet docs topic content added to this topic
* Link this topic to commands was handled in #1705

### CODE REVIEW

File: docs/standard/frameworks.md [hunk 1]
```diff
@@ -1,56 +1,293 @@
...
 
-The .NET ecosystem has a concept of frameworks. Frameworks define the API that you can use to target a particular platform. The .NET Framework 4.6 is one of those platforms. Frameworks are used in Visual Studio and other IDEs and editors to provide you with the correct set of APIs. They are also used by NuGet, for both production and consumption of NuGet packages, to ensure that you produce and use appropriate packages (and underlying assets) for the framework you are targeting. One can think of frameworks as one of the key currencies in the .NET ecosystem. The concept is there for correctness, to help you and your customers avoid seeing @System.MissingMethodException and friends at runtime.
+The .NET ecosystem has a concept of *target frameworks*. Frameworks define the API that you can use to target a particular platform. For example, the .NET Framework 4.6 is a platform that you can target. Visual Studio and other IDEs and editors use frameworks to provide you with the correct set of APIs. NuGet also uses them to ensure that you produce and consume appropriate NuGet packages and underlying assets for the framework you're targeting.
 
-## Framework Versions
+Refernece | Name
```
Comment (mairaw) [inline docs/standard/frameworks.md]: typo: Refernece

also, this feels a bit lost here without the explanation we had before.

File: docs/standard/frameworks.md [hunk 2]
```diff
@@ -1,56 +1,293 @@
...
+--- | ---
+Product | .NET Framework 4.6.2 or .NET 4.6.2
+Framework | `.NETFramework,Version=4.6.2` or `net462`
+Family | `.NETFramework` or `net`
 
-The table below defines the set of frameworks that you can use, how they are referred to and which version of the [.NET Standard Library](library.md) that they implement.
+## Lastest framework versions
```
Comment (mairaw) [inline docs/standard/frameworks.md]: typo: lastest

File: docs/standard/frameworks.md [hunk 3]
```diff
@@ -1,56 +1,293 @@
...
 
-| Framework | Latest Version | Target Framework Moniker (TFM) | Compact Target Framework Moniker (TFM) | .NET Standard Version | Metapackage |
-|:--------: | :--: | :--: | :--: | :--: | :--: | :--: |
-| .NET Standard | 1.6 | .NETStandard,Version=1.6 | netstandard1.6 | N/A | [NETStandard.Library](https://www.nuget.org/packages/NETStandard.Library)|
-| .NET Core Application | 1.0.1 | .NETCoreApp,Version=1.0 | netcoreapp1.0 | 1.6 | [Microsoft.NETCore.App](https://www.nuget.org/packages/Microsoft.NETCore.App)|
-| .NET Framework | 4.6.2 | .NETFramework,Version=4.6.2 | net462 | 1.5 | N/A |
+The table below defines the set of frameworks that you can use, how they're referenced, and which version of the [.NET Standard Library](library.md) they implement. These framework versions are the latest stable versions. Pre-release versions aren't shown.
```
Comment (mairaw) [inline docs/standard/frameworks.md]: The table below -> The following table (style guide rule)

File: docs/standard/frameworks.md [hunk 4]
```diff
@@ -1,56 +1,293 @@
...
+Universal Windows Platform | uap | uap [uap10.0]
+ | | uap10.0
+.NET Standard | netstandard | netstandard1.0
+ | | netstandard1.1
+ | | netstandard1.2
+ | | netstandard1.3
+ | | netstandard1.4
+ | | netstandard1.5
+ | | netstandard1.6
+.NET Core App | netcoreapp | netcoreapp1.0
```
Comment (mairaw) [inline docs/standard/frameworks.md]: we're missing netcoreapp1.1 here

**Comment by mairaw** [comment]
@terrajobst can you help review this? also, where can I find where the TFMs are defined? I wanted to double-check this list.

File: docs/standard/frameworks.md [hunk 5]
```diff
@@ -1,56 +1,293 @@
...
+| | wpa81
+| | netcore50
+win (Windows Store) | winrt
+| | winrt45
 
-**Referring to a family of Frameworks**
+## .NET Standard
 
-You can refer to a family of frameworks using long- or short-forms of the framework ID. Both are equally valid in the general case.
+The [.NET Standard](https://github.com/dotnet/standard) simplifies references between binary-compatible frameworks, allowing a single target framework to reference a combination of others. For background, see the [.NET Platform Guide](https://docs.microsoft.com/dotnet/articles/standard/index).
```
Comment (mairaw) [inline docs/standard/frameworks.md]: use relative link to topics inside the repo. and probably a better link is this: https://docs.microsoft.com/en-us/dotnet/articles/standard/library. I'll submit a PR to the NuGet docs for this.

File: docs/standard/frameworks.md [hunk 6]
```diff
@@ -1,56 +1,293 @@
...
 
-- `.NETFramework`
-- `net`
+The [NuGet Tools Get Nearest Framework Tool](https://aka.ms/s2m3th) simulates the NuGet logic used for the selection of one framework from many available framework assets in a package based on the project's framework. To use the tool, enter one project framework and one or more package frameworks. Click the **Submit** button. The tool indicates if the package frameworks you list are compatible with the project framework you provide.
+
+For NuGet 3.3 and earlier, use the `dotnet` series of monikers. For NuGet 3.4 and later, use the `netstandard` moniker syntax.
+
+## Portable Class Libraries
```
Comment (mairaw) [inline docs/standard/frameworks.md]: this one is a really long list and we don't even recommend people using it anymore. I think in this case we could link to the NuGet topic (https://docs.microsoft.com/nuget/schema/target-frameworks#portable-class-libraries)

File: docs/standard/frameworks.md [hunk 7]
```diff
@@ -1,56 +1,293 @@
 ---
-title: Frameworks and Targets
-description: Explains the concepts of framework targets when writing .NET code.
-keywords: .NET, .NET Core
+title: Target frameworks
```
Comment (mairaw) [inline docs/standard/frameworks.md]: .NET Target Frameworks | Microsoft Docs (no need to add the .NET for the H1)

**Review by mairaw — CHANGES_REQUESTED** [review]
a few comments

File: docs/standard/frameworks.md [hunk 8]
```diff
@@ -1,56 +1,293 @@
...
+ | | net35
+ | | net40
+ | | net403
+ | | net45
+ | | net451
+ | | net452
+ | | net46
+ | | net461
+ | | net462
+.NET Core | netcore | netcore [netcore45]
```
Comment (GuardRex) [inline docs/standard/frameworks.md]: Depreciated .NET Core TFM's? ... that haven't been formally moved to the "depreciated" list yet because some libs perhaps still use them??? *wilding guessing*

File: docs/standard/frameworks.md [hunk 9]
```diff
@@ -1,56 +1,143 @@
...
+Name | Abbreviation | TFMs/TxMs
+--- | --- | ---
+.NET Standard | netstandard | netstandard1.0
+ | | netstandard1.1
+ | | netstandard1.2
+ | | netstandard1.3
+ | | netstandard1.4
+ | | netstandard1.5
+ | | netstandard1.6
+.NET Core App | netcoreapp | netcoreapp1.0 | netcoreapp1.1
```
Comment (mairaw) [inline docs/standard/frameworks.md]: .netcoreapp1.1 should be in a new row

File: docs/standard/frameworks.md (hunk 8 above)
Comment (mairaw) [inline docs/standard/frameworks.md]: I'll wait on @terrajobst help on this one. I think it has to do with UWP.

File: docs/standard/frameworks.md [hunk 10]
```diff
@@ -1,56 +1,143 @@
 ---
-title: Frameworks and Targets
-description: Explains the concepts of framework targets when writing .NET code.
-keywords: .NET, .NET Core
+title: Target frameworks | Microsoft Docs
+description: Explains the concepts of target frameworks when writing .NET code.
+keywords: .NET, .NET Core, framework, tfm
```
Comment (mairaw) [inline docs/standard/frameworks.md]: tfm -> TFM

**Review by mairaw — CHANGES_REQUESTED** [review]
some minor comments. almost ready to go. I just want to clarify the netcore vs. netcoreapp TFMs more clearly. and I'm also trying to find out where these are defined so we can double check if the list is complete.

File: docs/standard/frameworks.md [hunk 11]
```diff
@@ -1,56 +1,144 @@
...
+
+## Latest framework versions
+
+The following table defines the set of frameworks that you can use, how they're referenced, and which version of the [.NET Standard Library](library.md) they implement. These framework versions are the latest stable versions. Pre-release versions aren't shown.
+
+Framework | Latest Version | Target Framework Moniker (TFM) | Compact Target Framework Moniker (TFM) | .NET Standard Version | Metapackage
+:--------: | :--: | :--: | :--: | :--: | :--: | :--:
+.NET Standard | 1.6.1 | .NETStandard,Version=1.6 | netstandard1.6 | N/A | [NETStandard.Library](https://www.nuget.org/packages/NETStandard.Library)
+.NET Core Application | 1.1.1 | .NETCoreApp,Version=1.1 | netcoreapp1.1 | 1.6 | [Microsoft.NETCore.App](https://www.nuget.org/packages/Microsoft.NETCore.App)
+.NET Framework | 4.6.2 | .NETFramework,Version=4.6.2 | net462 | 1.5 | N/A
```
Comment (svick) [inline docs/standard/frameworks.md]: This table doesn't seem to render properly on GitHub (probably caused by the recent GitHub change).

File: docs/standard/frameworks.md [hunk 12]
```diff
@@ -1,56 +1,144 @@
...
+| | wpa81
+| | netcore50
+win (Windows Store) | winrt
+| | winrt45
+
+## .NET Standard
+
+The [.NET Standard](https://github.com/dotnet/standard) simplifies references between binary-compatible frameworks, allowing a single target framework to reference a combination of others. For background, see the [.NET Standard Library](library.md).
+
+The [NuGet Tools Get Nearest Framework Tool](https://aka.ms/s2m3th) simulates the NuGet logic used for the selection of one framework from many available framework assets in a package based on the project's framework. To use the tool, enter one project framework and one or more package frameworks. Click the **Submit** button. The tool indicates if the package frameworks you list are compatible with the project framework you provide.
```
Comment (svick) [inline docs/standard/frameworks.md]: This seems to link to version 3.5.0-beta2-1484 of the tool, when versions 3.5.0 and 4.0.0 are available. Shouldn't the link be changed to one of those?

File: docs/standard/frameworks.md [hunk 13]
```diff
@@ -1,56 +1,144 @@
...
+
+For NuGet 3.3 and earlier, use the `dotnet` series of monikers. For NuGet 3.4 and later, use the `netstandard` moniker syntax.
```
Comment (svick) [inline docs/standard/frameworks.md]: Is information on how to use an old version of NuGet still useful?

File: docs/standard/frameworks.md [hunk 14]
```diff
@@ -1,56 +1,144 @@
...
+ monomac | Mono Support for OSX | netstandard1.4
+ xamarinios | Support for Xamarin for iOS | netstandard1.4
+ xamarinmac | Supports for Xamarin for Mac | netstandard1.4
+ xamarinpsthree | Support for Xamarin on Playstation 3 | netstandard1.4
+ xamarinpsfour | Support for Xamarin on Playstation 4 | netstandard1.4
+ xamarinpsvita | Support for Xamarin on PS Vita | netstandard1.4
+ xamarinwatchos | Xamarin for Watch OS | netstandard1.4
+ xamarintvos | Xamarin for TV OS | netstandard1.4
+ xamarinxboxthreesixty | Xamarin for XBox 360 | netstandard1.4
+ xamarinxboxone | Xamarin for XBox One | netstandard1.4
```
Comment (svick) [inline docs/standard/frameworks.md]: What does this table do here? It doesn't seem to be related to the paragraph right before it. Shouldn't it have some sort of text introduction?

File: docs/standard/frameworks.md (hunk 11 above)
Comment (GuardRex) [inline docs/standard/frameworks.md]: Probably the exterior pipes ... they aren't technically required, but I already had one table :boom: on me without them. I'll add them back here and see if that does the trick.

File: docs/standard/frameworks.md (hunk 12 above)
Comment (GuardRex) [inline docs/standard/frameworks.md]: Yes, great catch! I'll fix it. Let's just go with `http://nugettoolsdev.azurewebsites.net/`, since the latest framework is loaded :sparkles: auto-magically :sparkles:

File: docs/standard/frameworks.md (hunk 13 above)
Comment (GuardRex) [inline docs/standard/frameworks.md]: @mairaw ?

File: docs/standard/frameworks.md (hunk 14 above)
Comment (GuardRex) [inline docs/standard/frameworks.md]: I cut what @mairaw said to cut. @mairaw, do you want to cut this one?

Comment (mairaw) [inline docs/standard/frameworks.md]: yes, this is also part of the PCL section on the NuGet docs

File: docs/standard/frameworks.md (hunk 13 above)
Comment (mairaw) [inline docs/standard/frameworks.md]: I'd remove it since the dotnet ones already show up as deprecated earlier

File: docs/standard/frameworks.md (hunk 11 above)
Comment (GuardRex) [inline docs/standard/frameworks.md]: I found the prob with this ... it's not the exterior pipes ... the columns are literally wrong. I'll fix it.

**Comment by GuardRex** [comment]
Thanks @svick ... I think that licked it.

We decided to clean up tables generally as we go. *Future generations will thank us!*

File: docs/standard/frameworks.md [hunk 15]
```diff
@@ -1,56 +1,126 @@
...
-
-You can refer to a family of frameworks using long- or short-forms of the framework ID. Both are equally valid in the general case.
-
-- `.NETFramework`
-- `net`
+# Target frameworks
+
+The .NET ecosystem has a concept of *target frameworks*. Frameworks define the API that you can use to target a particular platform. For example, the .NET Framework 4.6 is a platform that you can target. Visual Studio and other IDEs and editors use frameworks to provide you with the correct set of APIs. NuGet also uses them to ensure that you produce and consume appropriate NuGet packages and underlying assets for the framework you're targeting.
+
+| Reference | Name                                      |
```
Comment (mairaw) [inline docs/standard/frameworks.md]: I thought I had commented it before but couldn't find it. This information seems lost here compared to how the information was presented before.

Comment (GuardRex) [inline docs/standard/frameworks.md]: I *did* make an attempt to rewrite it to make sense ... and failed :cry:. I'll pull up the text from my prior attempt and see where I left off with it.

When you say "how the information was presented before," that predates me here. This is the way I found it sans the two sentences at the end that I stripped off.

Comment (mairaw) [inline docs/standard/frameworks.md]: I thought this was you rewriting the Writing about Frameworks section which is shown on the diff, no?

**Comment by GuardRex** [comment]
@mairaw Oh, yes. I see what you mean now. I thought you were referring to the first paragraph. I brought back the *Writing about frameworks* section with upgrades. The table was meant simply to condense all of it, but I like this layout as well. I think we know who influenced my earlier decision *cough* *cough* *Rick* *cough*. :grinning:

**Comment by GuardRex** [comment]
@mairaw Oh, yes. I see what you mean now. I thought you were referring to the first paragraph. I brought back the *Writing about frameworks* section with upgrades. The table was meant simply to condense all of it, but I like this layout, too. I think we know who influenced my earlier decision *cough* *cough* *Rick* *cough*. :grinning:

I plopped that alternative frameworks language in here for feedback. cc/ @weshaggard @terrajobst

File: docs/standard/frameworks.md [hunk 16]
```diff
@@ -1,56 +1,149 @@
...
 
-## Framework Versions
+The terms *framework* and *platform* are sometimes confusing given how they're used in phrases. Making interpretation worse, the term *platform* has different meanings in different contexts. For example, you'll see ".NET Core" described as the ".NET Core framework" in the context of building apps and libraries and also described as the ".NET Core platform" in the context of where app and library code is executed. A *computing platform* describes *where and how* an application is run. Since .NET Core executes code with the [.NET Core Common Language Runtime (CoreCLR)](https://github.com/dotnet/coreclr), it's also a platform. The same is true of the .NET Framework, which has the [Common Language Runtime (CLR)](https://msdn.microsoft.com/library/8bs2ecf4(v=vs.110).aspx) to execute app and library code that was developed with the .NET Framework's framework objects, methods, and tools. You'll frequently see the term "cross-platform" in documentation; but when you see that term, you should think "cross-operating system and cross-architecture (x86, x64, arm)," because that's the meaning that the author usually intends to convey.
```
Comment (mairaw) [inline docs/standard/frameworks.md]: https://msdn.microsoft.com/library/8bs2ecf4(v=vs.110).aspx -> https://github.com/dotnet/docs/blob/master/docs/standard/clr.md

File: docs/standard/frameworks.md [hunk 17]
```diff
@@ -1,56 +1,149 @@
...
 
-| Framework | Latest Version | Target Framework Moniker (TFM) | Compact Target Framework Moniker (TFM) | .NET Standard Version | Metapackage |
-|:--------: | :--: | :--: | :--: | :--: | :--: | :--: |
-| .NET Standard | 1.6 | .NETStandard,Version=1.6 | netstandard1.6 | N/A | [NETStandard.Library](https://www.nuget.org/packages/NETStandard.Library)|
-| .NET Core Application | 1.0.1 | .NETCoreApp,Version=1.0 | netcoreapp1.0 | 1.6 | [Microsoft.NETCore.App](https://www.nuget.org/packages/Microsoft.NETCore.App)|
-| .NET Framework | 4.6.2 | .NETFramework,Version=4.6.2 | net462 | 1.5 | N/A |
+When you *target a framework* or target several of them, you've decided which set(s) of APIs and which version(s) of those APIs you would like to use. Frameworks are referenced in several ways: by product name, by long- or short-form framework names, and by family.
```
Comment (mairaw) [inline docs/standard/frameworks.md]: style guide rule about adding (s) to make plural: Don't add (s) or (es) to a singular noun to indicate that it can be singular or plural. Most of the time, use plural instead. If it's important to indicate both, use one or more.

File: docs/standard/frameworks.md [hunk 18]
```diff
@@ -1,56 +1,149 @@
...
+|                            |              | net45                                        |
+|                            |              | net451                                       |
+|                            |              | net452                                       |
+|                            |              | net46                                        |
+|                            |              | net461                                       |
+|                            |              | net462                                       |
+| Windows Store              | netcore      | netcore [netcore45]                          |
+|                            |              | netcore45 [win, win8]                        |
+|                            |              | netcore451 [win81]                           |
+| .NET MicroFramework        | netmf        | netmf                                        |
```
Comment (mairaw) [inline docs/standard/frameworks.md]: space between Micro and framework

**Review by mairaw — APPROVED** [review]
I've left a few last comments for you to address before merging but this is a big improvement IMO. Thanks @GuardRex!
